File: examples/governance/compliance_domain/lib/compliance_domain/training_record/events/completed_training.rb
module ComplianceDomain
  class TrainingRecord
    module Events
      class CompletedTraining
        attr_reader :aggregate_id, :training_record_id, :certification_id, :expires_at, :stakeholder_id, :policy_id, :completed_at, :status, :occurred_at

        def initialize(aggregate_id: nil, training_record_id: nil, certification_id: nil, expires_at: nil, stakeholder_id: nil, policy_id: nil, completed_at: nil, status: nil)
          @aggregate_id = aggregate_id
          @training_record_id = training_record_id
          @certification_id = certification_id
          @expires_at = expires_at
          @stakeholder_id = stakeholder_id
          @policy_id = policy_id
          @completed_at = completed_at
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/compliance_static_go/domain/completed_training.go
package domain

import "time"

type CompletedTraining struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	TrainingRecordId string `json:"training_record_id"`
	CertificationId string `json:"certification_id"`
	ExpiresAt time.Time `json:"expires_at"`
	StakeholderId string `json:"stakeholder_id"`
	PolicyId string `json:"policy_id"`
	CompletedAt time.Time `json:"completed_at"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e CompletedTraining) EventName() string { return "CompletedTraining" }

func (e CompletedTraining) GetOccurredAt() time.Time { return e.OccurredAt }
